File: VB/MultiRangeTrackBarControlExample/MultiRangeTrackBar/RepositoryItemMultiTrackBar.vb
Imports DevExpress.Accessibility
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository
Imports System.Drawing

Namespace MultiRangeTrackBarControlExample

    <UserRepositoryItem("RegisterCustomTrackBar")>
    Public Class RepositoryItemMultiTrackBar
        Inherits RepositoryItemTrackBar

        Shared Sub New()
            Call RegisterCustomTrackBar()
        End Sub

        Public Const CustomEditName As String = "MultiRangeTrackBar"

        Public Sub New()
        End Sub

        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return CustomEditName
            End Get
        End Property

        Public Shared Sub RegisterCustomTrackBar()
            Dim img As Image = Nothing
            Call EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(CustomEditName, GetType(MultiRangeTrackBar), GetType(RepositoryItemMultiTrackBar), GetType(MultiTrackBarViewInfo), New TrackBarPainter(), True, img, GetType(TrackBarAccessible)))
        End Sub

        Public Overrides Sub Assign(ByVal item As RepositoryItem)
            BeginUpdate()
            Try
                MyBase.Assign(item)
                Dim source As RepositoryItemMultiTrackBar = TryCast(item, RepositoryItemMultiTrackBar)
                If source Is Nothing Then Return
                DrawRanges = source.DrawRanges
            Finally
                EndUpdate()
            End Try
        End Sub

        Public Property DrawRanges As Boolean = True
    End Class
End Namespace

File: CS/MultiRangeTrackBarControlExample/MultiRangeTrackBar/RepositoryItemMultiTrackBar.cs
using DevExpress.Accessibility;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;
using System.Drawing;

namespace MultiRangeTrackBarControlExample {
    [UserRepositoryItem("RegisterCustomTrackBar")]
    public class RepositoryItemMultiTrackBar : RepositoryItemTrackBar {
        static RepositoryItemMultiTrackBar() {
            RegisterCustomTrackBar();
        }

        public const string CustomEditName = "MultiRangeTrackBar";

        public RepositoryItemMultiTrackBar() {
        }

        public override string EditorTypeName => CustomEditName;

        public static void RegisterCustomTrackBar() {
            Image img = null;
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(CustomEditName, typeof(MultiRangeTrackBar), typeof(RepositoryItemMultiTrackBar), typeof(MultiTrackBarViewInfo), new TrackBarPainter(), true, img, typeof(TrackBarAccessible)));
        }

        public override void Assign(RepositoryItem item) {
            BeginUpdate();
            try {
                base.Assign(item);
                RepositoryItemMultiTrackBar source = item as RepositoryItemMultiTrackBar;
                if (source == null) return;
                DrawRanges = source.DrawRanges;
            } finally {
                EndUpdate();
            }
        }

        public bool DrawRanges { get; set; } = true;
    }
}
